modules/core/include/opencv2/core/mat.hpp

```diff
@@ -2451,7 +2451,7 @@ class CV_EXPORTS UMat
     //!  <0 - a diagonal from the lower half)
     UMat diag(int d=0) const;
     //! constructs a square diagonal matrix which main diagonal is vector "d"
-    static UMat diag(const UMat& d);
+    static UMat diag(const UMat& d, UMatUsageFlags usageFlags = USAGE_DEFAULT);
```

Comment:
It makes sense to add compatibility overloads.

```.diff
-static UMat diag(const UMat& d, UMatUsageFlags usageFlags = USAGE_DEFAULT);
+static UMat diag(const UMat& d, UMatUsageFlags usageFlags /*= USAGE_DEFAULT*/);
+static UMat diag(const UMat& d) { return diag(d, USAGE_DEFAULT); }  // OpenCV 5.0: remove compatibility overload
```

Reply:
got it. I'll force push an update with this approach